packages/office-ui-fabric-react/src/components/Layer/Layer.tsx

```diff
@@ -63,4 +113,27 @@ export class Layer extends BaseComponent<ILayerProps, {}> {
         />
     );
   }
+
+  private _removeLayerElement() {
+    if (this._layerElement) {
+      this.props.onLayerWillUnmount();
+
+      ReactDOM.unmountComponentAtNode(this._layerElement);
+      this._layerElement.remove();
+      this._layerElement = undefined;
+      this._hasMounted = false;
+    }
+  }
+
+  private _getHost(): Node {
+    let { hostId } = this.props;
+    let doc = getDocument(this._rootElement);
+
+    if (hostId) {
+      return doc.getElementById(hostId);
```

Comment:
What if the element returned is null? It's unlikely and we expect them to pass in a good value but it could cause problems.

Also I think you need the conditional to be if (hostId && doc) to check for ssr.